core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -524,7 +526,18 @@ class GroupMetadataManager(brokerId: Int,
           val memRecords = fetchDataInfo.records match {
             case records: MemoryRecords => records
             case fileRecords: FileRecords =>
-              buffer.clear()
+              val sizeInBytes = fileRecords.sizeInBytes
+              val bytesNeeded = Math.max(config.loadBufferSize, sizeInBytes)
```

Comment:
A comment may be helpful to mention that because of the `minOneMessage` behavior, we may need to grow the buffer to ensure progress can be made.